ksqldb-streams/src/test/java/io/confluent/ksql/execution/streams/materialization/ks/KsMaterializedWindowTableIQv2Test.java

```diff
@@ -0,0 +1,306 @@
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class KsMaterializedWindowTableIQv2Test {
+
+  private static final Duration WINDOW_SIZE = Duration.ofMinutes(1);
+  private static final int PARTITION = 0;
+
+  private static final LogicalSchema SCHEMA = LogicalSchema.builder()
+      .keyColumn(ColumnName.of("K0"), SqlTypes.STRING)
+      .valueColumn(ColumnName.of("v0"), SqlTypes.STRING)
+      .build();
+
+  private static final GenericKey A_KEY = GenericKey.genericKey(0);
+  private static final GenericKey A_KEY2 = GenericKey.genericKey(1);
+  private static final GenericKey A_KEY3 = GenericKey.genericKey(1);
+
+  protected static final Instant NOW = Instant.now();
+
+  private static final Range<Instant> WINDOW_START_BOUNDS = Range.closed(
+      NOW,
+      NOW.plusSeconds(10)
+  );
+
+  private static final Range<Instant> WINDOW_END_BOUNDS = Range.closed(
+      NOW.plusSeconds(5).plus(WINDOW_SIZE),
+      NOW.plusSeconds(15).plus(WINDOW_SIZE)
+  );
+
+  private static final ValueAndTimestamp<GenericRow> VALUE_1 = ValueAndTimestamp
+      .make(GenericRow.genericRow("col0"), 12345L);
+  private static final ValueAndTimestamp<GenericRow> VALUE_2 = ValueAndTimestamp
+      .make(GenericRow.genericRow("col1"), 45678L);
+  private static final ValueAndTimestamp<GenericRow> VALUE_3 = ValueAndTimestamp
+      .make(GenericRow.genericRow("col2"), 987865L);
+
+  @Mock
+  private KafkaStreams kafkaStreams;
+  @Mock
+  private KsStateStore stateStore;
+  @Mock
+  private WindowStoreIterator<ValueAndTimestamp<GenericRow>> fetchIterator;
+  @Mock
+  private KeyValueIterator<Windowed<GenericKey>, ValueAndTimestamp<GenericRow>> keyValueIterator;
+  @Captor
+  private ArgumentCaptor<QueryableStoreType<?>> storeTypeCaptor;
+
+  @Mock
+  private StateQueryResult queryResult;
+  @Mock
+  private QueryResult<WindowStoreIterator<ValueAndTimestamp<GenericRow>>> fetchResult;
+  @Mock
+  private QueryResult<KeyValueIterator<Windowed<GenericKey>, ValueAndTimestamp<GenericRow>>> keyValueResult;
+
+  private KsMaterializedWindowTableIQv2 table;
+
+  @Before
+  public void setUp() {
+    table = new KsMaterializedWindowTableIQv2(stateStore, WINDOW_SIZE);
+
+    when(stateStore.getStateStoreName()).thenReturn("rocksdb-store");
+    when(stateStore.getKafkaStreams()).thenReturn(kafkaStreams);
+    when(stateStore.schema()).thenReturn(SCHEMA);
+  }
+
+  @SuppressWarnings("UnstableApiUsage")
+  @Test
+  public void shouldThrowNPEs() {
+    new NullPointerTester()
+        .setDefault(KsStateStore.class, stateStore)
+        .testConstructors(KsMaterializedWindowTable.class, Visibility.PACKAGE);
+  }
+
+  @Test
+  public void shouldThrowIfGettingStateStoreFails() {
+    // Given:
+    when(stateStore.getKafkaStreams()).thenThrow(new MaterializationTimeOutException("Boom"));
```

Comment:
It's hard to imagine that this method itself would throw an exception. Maybe we should make sure we get the right exception if the query itself throws and also if the query just returns a "failed" result?